rust/backend/daemon/src/features.rs

```diff
@@ -1,16 +1,230 @@
+                Some("AddGlobalRef"),
+                offset,
+                target,
+                None
+            )?;
+            self.trace_add_global = Some(jni_add_global.take_link(link_id)?);*/
+        }
+
+        if self.trace_add_global.is_none() {
+            let jni_add_global: &mut UProbe = ebpf
+                .program_mut("trace_add_global")
+                .ok_or(EbpfError::ProgramError(
+                    aya::programs::ProgramError::InvalidName {
+                        name: "trace_add_global".to_string(),
+                    },
+                ))?
+                .try_into()?;
+
+            let offset = todo!("get offset of symbol");
+            let target = todo!("get absolute path to library/ binary");
+
+            /*let link_id = jni_add_global.attach(
+                Some("AddGlobalRef"),
+                offset,
+                target,
+                None
+            )?;
+            self.trace_add_global = Some(jni_add_global.take_link(link_id)?);*/
+        }
+
+        if self.trace_del_global.is_none() {
+            let jni_del_global: &mut UProbe = ebpf
+                .program_mut("trace_del_global")
+                .ok_or(EbpfError::ProgramError(
+                    aya::programs::ProgramError::InvalidName {
+                        name: "trace_del_global".to_string(),
+                    },
+                ))?
+                .try_into()?;
+
+            let offset = todo!("get offset of symbol");
+            let target = todo!("get absolute path to library/ binary");
+
+            /*let link_id = jni_del_global.attach(
+                Some("DeleteGlobalRef"),
+                offset,
+                target,
+                None
+            )?;
+            self.trace_del_global = Some(jni_del_global.take_link(link_id)?);*/
+        }
+
+        Ok(())
+    }
+
+    pub fn detach(&mut self, ebpf: &mut Ebpf) -> Result<(), EbpfError> {
+        let _ = self.trace_add_local.take();
+        let _ = self.trace_del_local.take();
+        let _ = self.trace_add_global.take();
+        let _ = self.trace_del_global.take();
+
+        Ok(())
+    }
+
+    fn update_pids(
+        &mut self,
+        ebpf: &mut Ebpf,
+        pids: &[u32]
+    ) -> Result<(), EbpfError> {
+        let mut pids_to_track: HashMap<_, u32, u32> = ebpf
+            .map_mut("JNI_REF_PIDS")
+            .ok_or(EbpfError::MapError(aya::maps::MapError::InvalidName {
+                name: "JNI_REF_PIDS".to_string(),
+            }))?
+            .try_into()?;
+
+        let new_keys = pids.iter().copied().collect::<BTreeSet<u32>>();
+        let existing_keys = pids_to_track.keys().collect::<Result<BTreeSet<u32>, _>>()?;
+
+        for key_to_remove in existing_keys.difference(&new_keys) {
+            pids_to_track.remove(key_to_remove)?;
+        }
+
+        for key_to_add in new_keys.difference(&existing_keys) {
+            pids_to_track.insert(key_to_add, 0, 0)?;
+        }
+
+        Ok(())
+    }
```

Comment:
We should probably unify this kind of action, this is basically verbatim the same as for others.

Reply:
@ffranzgitHub is currently refactoring this file. I told him to extend his work to this function.